versioned/persist/tests/src/main/java/org/projectnessie/versioned/persist/tests/AbstractTieredCommitsTest.java

```diff
@@ -408,4 +375,65 @@ void recreateDefaultBranch() throws Exception {
       super(databaseAdapter);
     }
   }
+
+  @Test
+  void keyPrefixBasic(
+      @NessieAdapter @NessieAdapterConfigItem(name = "key.prefix", value = "foo")
+          DatabaseAdapter foo,
+      @NessieAdapter @NessieAdapterConfigItem(name = "key.prefix", value = "bar")
+          DatabaseAdapter bar)
+      throws Exception {
+
+    BranchName main = BranchName.of("main");
+
+    foo.commit(
+        ImmutableCommitAttempt.builder()
+            .commitToBranch(main)
+            .commitMetaSerialized(ByteString.copyFromUtf8("meta-foo"))
+            .addPuts(
+                KeyWithBytes.of(
+                    Key.of("foo"), ContentsId.of("foo"), (byte) 0, ByteString.copyFromUtf8("foo")))
+            .build());
+    bar.commit(
+        ImmutableCommitAttempt.builder()
+            .commitToBranch(main)
+            .commitMetaSerialized(ByteString.copyFromUtf8("meta-bar"))
+            .addPuts(
+                KeyWithBytes.of(
+                    Key.of("bar"), ContentsId.of("bar"), (byte) 0, ByteString.copyFromUtf8("bar")))
+            .build());
+
+    Hash fooMain = foo.toHash(main);
+    Hash barMain = bar.toHash(main);
+
+    Hash fooBranch = foo.create(BranchName.of("foo-branch"), fooMain);
+    Hash barBranch = bar.create(BranchName.of("bar-branch"), barMain);
+
+    assertThat(fooMain).isNotEqualTo(barMain).isEqualTo(fooBranch);
+    assertThat(barMain).isNotEqualTo(fooMain).isEqualTo(barBranch);
+
+    // Verify that key-prefix "foo" only sees "its" main-branch and foo-branch
+    assertThat(foo.namedRefs())
+        .containsExactlyInAnyOrder(
+            WithHash.of(fooMain, main), WithHash.of(fooBranch, BranchName.of("foo-branch")));
+    assertThat(bar.namedRefs())
+        .containsExactlyInAnyOrder(
+            WithHash.of(barMain, main), WithHash.of(barBranch, BranchName.of("bar-branch")));
+
+    assertThatThrownBy(() -> foo.commitLog(barBranch))
+        .isInstanceOf(ReferenceNotFoundException.class);
+    assertThatThrownBy(() -> bar.commitLog(fooBranch))
+        .isInstanceOf(ReferenceNotFoundException.class);
+
+    try (Stream<CommitLogEntry> log = foo.commitLog(fooBranch)) {
+      assertThat(log.collect(Collectors.toList()))
+          .extracting(CommitLogEntry::getMetadata)
+          .containsExactlyInAnyOrder(ByteString.copyFromUtf8("meta-foo"));
```

Comment:
maybe extract `ByteString.copyFromUtf8("meta-foo")` into a variable. Same for `meta-bar`